### ISSUE TITLE
Teoz: Shape of note changes only when "over" is used

### ISSUE THREAD

**Issue by danny-mhlv** [plantuml/plantuml#1451]
**Describe the bug**
When using `!pragma teoz true` - shape of the note changes only when "over" position is used.

**To Reproduce**

1. Create basic sequence diagram (I've used PlantUML Web Server)
2. Add `!pragma teoz true`
3. Add `hnote` for example (same thing for `rnote`)
4. Position it on the `left`
5. See that note does not change it's shape
6. Now position the note `over` a participant for example
7. Now the shape's changed

Code for image 1, when shape stays default.
```plantuml
@startuml
!pragma teoz true
Bob -> Alice : hello
hnote left: text
@enduml
```

Code for image 2, when shape is changed.
```plantuml
@startuml
!pragma teoz true
Bob -> Alice : hello
hnote over Bob: text
@enduml
```

**Expected behavior**

**Screenshots**
Image 1.
![Screenshot from 2023-06-11 02-33-23](https://github.com/plantuml/plantuml/assets/44134636/8a46b3bc-c4d9-4d37-b500-3501d0995719)
Image 2.
![Screenshot from 2023-06-11 02-39-27](https://github.com/plantuml/plantuml/assets/44134636/f9d44d7a-e289-4325-a834-5b2173f4eab0)

**Desktop:**

- OS: Ubuntu 22.04.2 LTS
- Browser Firefox v113.0.2

**Comment by chipbite** [comment]
Propose to change title of issue to the wording you used in the description, e g:

> shape of the note changes only when "over" position is used.

So perhaps: 
# Teoz: Shape of note changes only when "over" used

**Comment by The-Lum** [comment]
_[Just for the record]_
We can also add some label _(here `m:teoz`)_.
_Ref._: [QA-17257](https://forum.plantuml.net/17257/issue-tracking?show=17315#c17315)

**Comment by jimnelson372** [comment]
It looks like the problem isn't with the "over", but with the the cases where "left" or "right" on hnote or rnote does not specify a particular participant.  Here is a script that tries all the combinations (i.e. left, right, over Bob; left,right,over Alice, left, right without specific participant).  

```
@startuml
!pragma teoz true

Bob -> Alice : hello
rnote right: rnote\n just right
rnote left: rnote\n just left

Bob <- Alice: hello
hnote right: hnote\n just right
hnote left: hnote\n just left

hnote over Alice: hnote\n over \n Alice
hnote left Alice: hnote\n left \n Alice
hnote right Alice: hnote\n right \n Alice
rnote over Alice: rnote\n over \n Alice
rnote left Alice: rnote\n left \n Alice
rnote right Alice: rnote\n right \n Alice

hnote over Bob: hnote\n over \n Bob
hnote left Bob: hnote\n left \n Bob
hnote right Bob: hnote\n right \n Bob
rnote over Bob: rnote\n over \n Bob
rnote left Bob: rnote\n left \n Bob
rnote right Bob: rnote\n right \n Bob
@enduml
```

Notice in the following generated diagram that it's the first four instances, where no participant is specified, where rnote and hnote revert back to the normal note shape.

![testteoz](https://github.com/plantuml/plantuml/assets/66284321/f5b8962b-850c-4fff-affa-cf56747a9447)

If Teoz is not activated, Puma has the correct shapes on all positions:

![testteoz](https://github.com/plantuml/plantuml/assets/66284321/07833b47-b237-44e5-8336-8507f9be3323)

I haven't yet looked into the code.  I was just looking at the example to understand the issue.

Regards,
Jim Nelson

**Comment by jimnelson372** [comment]
Okay, it turns out to be "message" notes that were only showing the normal note style.  The ones that named a participant to be left, over and right of would correctly show the Hexagon (hnote) and Box (rnote) styles.  I have a test that now shows it working.   I added one more message after the first two that just uses the traditional note style to demonstrate that still works:

![testteoz_rnote_hnote](https://github.com/plantuml/plantuml/assets/66284321/5ce3560f-8d0a-4051-b977-b02585ef301f)

My test script is now:

```
@startuml
!pragma teoz true

Bob -> Alice : hello
rnote right: rnote\n just right
rnote left: rnote\n just left

Bob <- Alice: hello
hnote right: hnote\n just right
hnote left: hnote\n just left

hnote over Alice: hnote\n over \n Alice
hnote left Alice: hnote\n left \n Alice
hnote right Alice: hnote\n right \n Alice
rnote over Alice: rnote\n over \n Alice
rnote left Alice: rnote\n left \n Alice
rnote right Alice: rnote\n right \n Alice

hnote over Bob: hnote\n over \n Bob
hnote left Bob: hnote\n left \n Bob
hnote right Bob: hnote\n right \n Bob
rnote over Bob: rnote\n over \n Bob
rnote left Bob: rnote\n left \n Bob
rnote right Bob: rnote\n right \n Bob

Bob -> Alice : hello
note right: note\n just right
note left: note\n just left
@enduml
```

Let me know how you if this looks good to you. :)

Regards,

Jim Nelson

**Comment by The-Lum** [comment]
Hi @jimnelson372, 

That seems good. 👍 

But:
- Could you just add this test? 
```puml
note across: note\nacross
hnote across: hnote\nacross
rnote across: rnote\nacross
```

to now cover all the cases:
```puml
@startuml
!pragma teoz true

Bob -> Alice : hello
rnote right: rnote\n just right
rnote left: rnote\n just left

Bob <- Alice: hello
hnote right: hnote\n just right
hnote left: hnote\n just left

hnote over Alice: hnote\n over \n Alice
hnote left Alice: hnote\n left \n Alice
hnote right Alice: hnote\n right \n Alice
rnote over Alice: rnote\n over \n Alice
rnote left Alice: rnote\n left \n Alice
rnote right Alice: rnote\n right \n Alice

hnote over Bob: hnote\n over \n Bob
hnote left Bob: hnote\n left \n Bob
hnote right Bob: hnote\n right \n Bob
rnote over Bob: rnote\n over \n Bob
rnote left Bob: rnote\n left \n Bob
rnote right Bob: rnote\n right \n Bob

Bob -> Alice : hello
note right: note\n just right
note left: note\n just left

note across: note\nacross
hnote across: hnote\nacross
rnote across: rnote\nacross
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/RPDT3e8m48JVESLw02v0Z47FuKl5IZ65cctnmTDRMaf_ja-4cIyp6RBqsZ0qwooRqmVPE3CmNFt0uCgRvgOUq5xWAgU1GmU2IwaQN9JXWDCeJ0VkvRx0UzN6Qxil-QkqDybddbkV6I956Ybef4WY1OtqjlfoF69to6dswUGafkISLd2kkWIzU918ch7QZAGPgyr8cx7Ud8wsdvHEjc8wE6D2QMHyUWOTXHk5HLivCpAnhPmOcQIDpDke-k6bHq9l93aHUY7EP0CghOFjt_RvmHAPXvc7cTVpvMd_cZy0)](https://www.planttext.com/?text=RPDT3e8m48JVESLw02v0Z47FuKl5IZ65cctnmTDRMaf_ja-4cIyp6RBqsZ0qwooRqmVPE3CmNFt0uCgRvgOUq5xWAgU1GmU2IwaQN9JXWDCeJ0VkvRx0UzN6Qxil-QkqDybddbkV6I956Ybef4WY1OtqjlfoF69to6dswUGafkISLd2kkWIzU918ch7QZAGPgyr8cx7Ud8wsdvHEjc8wE6D2QMHyUWOTXHk5HLivCpAnhPmOcQIDpDke-k6bHq9l93aHUY7EP0CghOFjt_RvmHAPXvc7cTVpvMd_cZy0) 

Regards,
Th.

**Comment by jimnelson372** [comment]
Yes, @The-Lum, the across still works.  I can see why you wanted to see this, since I had stressed that the problem was where there was not **named participants**, yet the `across` doesn't have named participants either.   

The real issue is whether the note is considered to be connected to a message, or to one or all of the participants.

`rnote across : Some text message`

is a note that is connected to all the participants.

`rnote left Bob: Some text message`

is a note (box style) related to a particular participant: Bob.  While:

`rnote left : Some text message`

is a note (box style) related to the message it follows.  It's on this last kind of note where it is currently only displaying the normal style.

Here is the result of your script, with the fixes:

![testteoz_rnote_hnote](https://github.com/plantuml/plantuml/assets/66284321/e646203e-fcda-4187-a94b-12f9a96a5298)

Thanks for the request.

Jim N.

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 
This is now fixed on [V1.2024.6](https://github.com/plantuml/plantuml/releases/tag/v1.2024.6).

- Thanks @danny-mhlv for the report.
- Thanks @jimnelson372  for your work.

Regards.
